components/codeblock/codeblock.mdx

```diff
@@ -0,0 +1,51 @@
+import { Canvas, Story, Subtitle, ArgsTable, PRIMARY_STORY } from '@storybook/addon-docs/blocks';
+import DtCodeblock from './codeblock';
+
+# Codeblock
+
+<Subtitle>
+  A codeblock is a UI element intended to display a piece of code in a separate and self-contained block.
+</Subtitle>
+
+## Base Style
+
+Text passed into the component is rendered under the assumption that it's preformatted.
+
+For example, to render something like:
+
+```js
+function someFunction() {
+  return "some result";
+```
+
+You'll have to do:
+
+```vue
+<dt-codeblock
+  text="'function someFunction() {\n  return "some result";\n}'"
+/>
+```
+
+<Canvas>
+  <Story id="components-codeblock--default" />
+</Canvas>
+
+## Slots, Props & Events
+
+<ArgsTable story={PRIMARY_STORY} />
+
+## Usage
+
+### Import
+
+```jsx
+import { DtCodeblock } from '@dialpad/dialtone-vue';
+```
+
+### Default
+
+```vue
+<dt-codeblock
+  text="'some pre-formatted codeblock'"
```

Comment:
the inner quotes aren't necessary right?

Reply:
You're right, good catch! I'll change this